extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -436,6 +436,57 @@ class LifecycleTask extends BackbeatTask {
         });
     }
 
+    /**
+     * Check that at least one rule has a tag or more.
+     * @param {array} rules - array of rules
+     * @return {boolean} true if at least one rule has a tag, false otherwise
+     */
+    _rulesHaveTag(rules) {
+        return rules.some(rule => {
+            if (!rule.Filter) {
+                return false;
+            }
+
+            const tags = rule.Filter.And
+                ? rule.Filter.And.Tags
+                : (rule.Filter.Tag && [rule.Filter.Tag]);
+            return tags && tags.length > 0;
```

Comment:
I know this can be shorter / easier with null chaining but can't seem to get all edge cases at the moment...

Reply:
```suggestion
            return (rule?.Filter?.And?.Tags || []).length > 0 ||
                rule?.Filter?.Tag !== undefined;
```
Ok best I got...